ClassLibrary/ClassLibrary.Test/TestDepartmentReaders.cs

```diff
@@ -82,5 +82,48 @@ public void BorrowedBookNullArgument()
             Should.Throw<ArgumentNullException>(() => DR.BorrowBook(reader, null));
         }
 
+        [Test]
+        public void NotList_GetTimeWhenFreeBook()
+        {
+            DepartmentReaders DR = new DepartmentReaders();
+            //I think it doesn't matter which parameter returns. What do you think? 
+            Should.Throw<ArgumentNullException>(() => DR.GetTimeWhenFreeBook(null));
+            Should.Throw<ArgumentNullException>(() => DR.GetTimeWhenFreeBook(new System.Collections.Generic.List<BorrowedBook>()));
+        }
+
+        [Test]
+        public void ReturnNotExistBook()
+        {
+            DepartmentReaders DR = new DepartmentReaders();
+            Author author = new Author("Lev", "Tolstoj");
+            Book book = new Book("226611156", "War and Peace", author);
+            Reader reader = new Reader("Lev", "Tolstoj");
+            DR.AddReader(reader);
+
+            Should.Throw<ArgumentException>(() => DR.ReturnBook(reader.ID, "Mu-Mu")).Message.ShouldBe("you didn't take this book");
+        }
+
+        [Test]
+        public void ReturnNotSetBook()
+        {
+            DepartmentReaders DR = new DepartmentReaders();
+            Author author = new Author("Lev", "Tolstoj");
+            Book book = new Book("226611156", "War and Peace", author);
+            Reader reader = new Reader("Lev", "Tolstoj");
+            DR.AddReader(reader);
+
+            Should.Throw<ArgumentNullException>(() => DR.ReturnBook(reader.ID, "")).ParamName.ShouldBe("NameBook");
+        }
+
+        [Test]
+        public void ReturnBookNotExistReader()
+        {
+            DepartmentReaders DR = new DepartmentReaders();
+            Author author = new Author("Lev", "Tolstoj");
```

Comment:
you don't need to create `author`, `book` and `reader` here

Reply:
+